### PR TITLE
ROLLO Verification Test 

### LINKED ISSUE #11: Integration test to test if default genetic algorithm can reach a correct verified solution

### PR DESCRIPTION
**Pull request by gwenchee**
The Ackley Function is one of the well-known benchmarks used for evolutionary or metaheuristic optimization. I use this function to test if the custom evolutionary algorithm in ROLLO reaches its minimum value. 

More information about the Ackley function: https://www.sfu.ca/~ssurjano/ackley.html 

Closes #11

### CODE REVIEW

**Comment by smpark7** [comment]
I created a clean conda environment, installed openmc from conda-forge, and then built rollo from source. When I ran the tests, I got the following failed tests:
```========================================================= short test summary info ==========================================================
FAILED test_evaluation.py::test_eval_fn_generator -   File "<unknown>", line 0
FAILED test_evaluation.py::test_get_output_vals - FileNotFoundError: [Errno 2] No such file or directory: './input_test_files'
FAILED test_evaluation.py::test_render_jinja_template_python - FileNotFoundError: [Errno 2] No such file or directory: './input_test_files'
FAILED test_executor.py::test_load_evaluator - FileNotFoundError: [Errno 2] No such file or directory: './input_test_files'
FAILED test_integration_ackley_minimum.py::test_ackley_minimum_check - assert 0.13762902276411682 < 0.1
FAILED test_openmc_evaluation.py::test_evaluate_keff - FileNotFoundError: [Errno 2] No such file or directory: './input_test_files/test_e...
================================================ 6 failed, 26 passed, 32 warnings in 8.50s =================================================
```
It seems that your ackley test sometimes fails, maybe because it's not running enough generations or your assertion is too strict. 

PS. I'm running Ubuntu 20.04

File: tests/test_integration_ackley_minimum.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    toolbox.register("x1", random.uniform, -32.768, 32.768)
+    toolbox.register("x2", random.uniform, -32.768, 32.768)
+    toolbox.pop_size = 1000
+    toolbox.ngen = 20
+
+
+    def ind_vals():
+        x1 = toolbox.x1()
+        x2 = toolbox.x2()
+        return creator.Ind([x1, x2])
```
Comment (smpark7) [inline tests/test_integration_ackley_minimum.py:33]: I think you can define this function outside `test_ackyl_minimum_check()`

File: tests/test_integration_ackley_minimum.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
+"""Integration Test to validate the Evolutionary Algorithm.
+The Ackley function is one of the well-known benchmarks used for evolutionary
+or metaheuristic optimization.
```
Comment (smpark7) [inline tests/test_integration_ackley_minimum.py:3]: Add a reference or link here about the Ackley function

File: tests/test_integration_ackley_minimum.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    def evaluator_fn(ind):
+        x1, x2 = ind[0], ind[1]
+        ackley = (
+            -20 * np.exp(-0.2 * np.sqrt(1 / 2 * (x1 ** 2 + x2 ** 2)))
+            - np.exp(1 / 2 * (np.cos(2 * np.pi * x1) + np.cos(2 * np.pi * x2)))
+            + 20
+            + np.exp(1)
+        )
+        return tuple([ackley])
```
Comment (smpark7) [inline tests/test_integration_ackley_minimum.py:71]: I think you can define this outside too

Comment (gwenchee) [reply tests/test_integration_ackley_minimum.py:71]: I'm going to leave it, I was getting some errors when I keep it outside.

File: tests/test_integration_ackley_minimum.py (hunk 1 above)
Comment (gwenchee) [reply tests/test_integration_ackley_minimum.py:33]: I'm going to leave it, I was getting some errors when I keep it outside.

**Comment by gwenchee** [comment]
@smpark7 Thanks for the review. I'm not sure why you are getting ` No such file or directory: './input_test_files'` error. I have not encountered it on my ubuntu or my mac. Could you try on another machine?

**Comment by gwenchee** [comment]
Interesting... my CI got the same error as you @smpark7

**Comment by gwenchee** [comment]
@smpark7 I did not include test_evaluation.py in my CI currently so it passes the test. Let's merge this and I'll deal with the CI issues outlined in #16 in a different PR

**Comment by smpark7** [comment]
Sure, the Ackley test now passes for me too.

**Pull request merged**